mibidata/mibi_image.py

```diff
@@ -232,7 +278,16 @@ def __str__(self):
 
     def metadata(self):
         """Returns a dictionary of the image's metadata."""
-        return {key: getattr(self, key) for key in _ATTRIBUTES}
+        metadata_keys = [key for key in _EXPECTED_METADATA_ATTRIBUTES]
+        # find user-defined metadata
+        metadata_keys.extend(self._user_defined_attributes())
+        return {key: getattr(self, key) for key in metadata_keys}
+
+    def _user_defined_attributes(self):
+        """Returns a list of keys corresponding to the user-defined metadata."""
+        return [key for key in self.__dict__
+                if key not in _EXPECTED_METADATA_ATTRIBUTES and
+                key not in _NON_METADATA_ATTRIBUTES]
```

Comment:
Instead of this, which as you've pointed out will include any attributes added to the instance after construction, I think we should store a dict of the kwargs remaining after popping the required ones and use that as the user-defined attributes. 

I'm also open to having a method where a user can add more attributes later, but I'd rather it be an explicit setter than just grabbing anything from the object. This could be something we do in a subsequent ticket.

Reply:
@rfinck I understand your concern; but I implemented the dictionary on the 1st place, and @jaytarolli didn't like the solution. I now implemented a solution that I think should solve all conflicts:
 - all remaining kwargs (after popping the required attributes) are now added via `setattr` to the `MibiImage` instance, so that they can be retrieved as usual `image.personalized_attr`.
 - at the same time, I created a protected member `_user_defined_attributes` to store a list of keys in order to keep track of the user-defined attributes, otherwise it's very hard to pass them further to the tiff creator.
Let me know if this doesn't work for you.

Reply:
EDIT: please consider that pylint now complains if you try to access the user-defined metadata, for example:
`Instance of 'MibiImage' has no 'mass_range' member (no-member)`.